feat(authz): [FC-0099] create an custom error boundery

dcoa opened the pull request:
# Create a custom error boundary for libraries team management workflow

This PR creates a custom error boundary that maps errors `401/403/404/5xx`

## Changes overview

- Add a env variable to performs redirections to authoring MFE libraries
- Create a reload button for 5xx error that will retry the query
- Create a custom error to display error access denied when the user do not have permissions over a library (the validation permission endpoint does not performs that)


## Evidence

<img width="1144" height="765" alt="image" src="https://github.com/user-attachments/assets/4a9f6035-66fd-48e7-9bf1-bd8deb213e25" />

<img width="1144" height="765" alt="image" src="https://github.com/user-attachments/assets/4e316d9f-cae3-402d-a4f0-ec7e6852bb84" />

arbrandes commented on src/authz-module/libraries-manager/ErrorPage/index.tsx:
```diff
@@ -0,0 +1,95 @@
+    resetErrorBoundary();
+  };
+  return (
+    <Container className="d-flex flex-column align-items-center justify-content-center min-vh-100 bg-light-200" data-testid="error-page">
```
```suggestion
    <Container className="d-flex flex-column align-items-center justify-content-center min-vh-100 bg-light-200">
```

Do you foresee a need to use `getByTestId`?  If not, I'd rather we only add it if necessary ([for the usual reasons](https://testing-library.com/docs/queries/about/#priority)).

arbrandes reviewed (changes requested):
Looks good, though I did have one nit.

jacobo-dominguez-wgu commented on src/authz-module/libraries-manager/ErrorPage/index.tsx:
```diff
@@ -0,0 +1,95 @@
+  });
+};
+
+const librariesUrl = () => `${getConfig().AUTHORING_BASE_URL}/authoring/libraries`;
```
I am not getting why this one is a function and not just a constant.
```suggestion
const LIBRARIES_URL = `${getConfig().AUTHORING_BASE_URL}/authoring/libraries`;
```
And maybe it can be moved to the constants file.

jacobo-dominguez-wgu commented on src/authz-module/libraries-manager/ErrorPage/index.tsx:
```diff
@@ -0,0 +1,95 @@
+  } = getErrorConfig({ errorMessage, errorStatus });
+
+  const handleReload = () => {
+    if (reloading) { return; }
```
This line maybe not needed since the button is disabled when reloading is true.

jacobo-dominguez-wgu reviewed (commented):
I have not tested on my local yet, but overall it looks good. I have added a couple of comments with some suggestions let me know your thoughts. Nice work!

dcoa commented:
@jacobo-dominguez-wgu @arbrandes  I addressed your comments, and I just change the env variable to `COURSE_AUTHORING_MICROFRONTEND_URL` to avoid duplicate values in tutor-mfe for example.

dcoa commented:
@jacobo-dominguez-wgu we have a sandbox now https://openedx.atlassian.net/wiki/spaces/OEPM/pages/5279023110/AuthZ+Stage+Information
